### PR TITLE
Use Quay token and host instead of config file

### PR DESCRIPTION
**Pull request by ghost**
Use axios instead of request, because request is deprecated
Remove unnecessary dependencies
Use token and quay host instead of config file
Appsody generated file
Update README

### CODE REVIEW

**Comment by ghost** [comment]
@ParadoxGuitarist Please review

File: README.md [hunk 1]
```diff
@@ -5,20 +5,14 @@ This software provides middleware for self hosted instances of
...
-an error if this is not configured properly.
-
-You should also note that when using the "test"  button while setting up the
-webhook from Bitbucket, do not include testing the quay server link in the
-config.json, or the required secret configuration.
 
 ### Environmental Variables
 These need to be set when your final appsody container is deployed.
 
 `SERVER_SECRET` Required - Also needs to be set in your Bitbucket webhook.
```
Comment (enov) [inline README.md]: The `SERVER_SECRET` is no longer required and should be removed.

File: app.js [hunk 2]
```diff
@@ -3,16 +3,21 @@ module.exports = (/*options*/) => {
   const express = require('express');
   const app = express();
   const bodyParser = require('body-parser');
-  const config = require('./config.json');
-  const request = require('request');
+  const https = require('https');
+  const axios = require('axios');
   const crypto = require('crypto');
   const fs = require('fs');
+  const quayHost = process.env.QUAY_HOST;
```
Comment (enov) [inline app.js:14]: Do we need to validate and throw with a message that this is required?

File: app.js [hunk 3]
```diff
@@ -39,23 +44,17 @@ module.exports = (/*options*/) => {
...
-        agentOptions: {
-          ca: fs.readFileSync(`/project/user-app/${process.env.QUAY_CA_FILE}`)
-        }
-      }
-    } else {
-      json: quayoptions
-    }
-    request.post(config[payload.repository.name], quayoptions, (error, res, body) => {
+    // Make the POST request to the request url.
+    instance.post('https://' + quayHost + req.originalUrl, quaypayload, {
```
Comment (enov) [inline app.js]: Should we use the incoming `url` here instead of `req.originalUrl`? Otherwise why add and pass in `url`?

File: app.js [hunk 4]
```diff
@@ -39,23 +44,17 @@ module.exports = (/*options*/) => {
...
+            auth: {
+                    username: '$token',
+                    password: req.query.token
```
Comment (enov) [inline app.js]: If we do not want to use the `req` obj, we also need to pass a `token` var. See comment above.

**Review by enov — CHANGES_REQUESTED** [review]
Awesome job @SWheeler17 ! I left a couple of comments, kindly review, and address them if applicable.

**Review by enov — APPROVED** [review]
LGTM!

File: app.js (hunk 2 above)
Comment (ParadoxGuitarist) [reply app.js:14]: Yeah, we probably do.

**Comment by ParadoxGuitarist** [comment]
We need to edit lines 32-36 in the README.md since it references the SERVER_SECRET, which is deprecated with this PR. We should probably add some direction on how to build the URI based off of the BitBucket settings as some people may find it confusing.

File: README.md [hunk 5]
```diff
@@ -15,6 +15,14 @@ These need to be set when your final appsody container is deployed.
 need to set this to the name of the cert.pem that you've added to the root of
 this project. (example: QUAY_CA_FILE=ExampleRootCA.pem)
 
+### How it Works
```
Comment (enov) [inline README.md:18]: I appreciate this section in the README.

File: README.md [hunk 6]
```diff
@@ -15,6 +15,14 @@ These need to be set when your final appsody container is deployed.
...
+The application will forward your request to your `QUAY_HOST` environment
+variable i.e.
+https://your.initial.url/webhooks/push/trigger/id?token=token
+will forward to:
+https://your.quay.url/webhooks/push/trigger/id?token=token
```
Comment (enov) [inline README.md]: @SWheeler17 Could you kindly review this. We're converting the token query string to a basic auth format:
```
https://$token:token@your.quay.url/webhooks/push/trigger/id
```
^ Please double check

File: README.md [hunk 7]
```diff
@@ -32,8 +40,7 @@ Here are the steps you'll take:
 You'll need to add a webhook that points to your newly built container image,
 either with the FQDN or the IP address and then add the `/api` Endpoint.
 Example: https://192.168.1.101:3000/api or
-https://bitbucket-quay-connector.cloudapps.master.ose/api After that be sure to
-add the string you set for the `SERVER_SECRET`.
+https://bitbucket-quay-connector.cloudapps.master.ose/api. The server secret does not need to be configured.
```
Comment (enov) [inline README.md]: Mentioning server secret will confuse people. We do not need to say:
> The server secret does not need to be configured.

Once this is merged, the `SERVER_SECRET` env variable is only available in git history.

**Comment by enov** [comment]
@SWheeler17  thank you for your hard work on this.

**Comment by enov** [comment]
LGTM cc @ParadoxGuitarist

**Pull request merged**